File: screenshot.mjs
import puppeteer from 'puppeteer';
import { fileURLToPath } from 'url';
import path from 'path';
import fs from 'fs';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const htmlFile = path.join(__dirname, 'index.html');
const url = `file://${htmlFile}`;

const browser = await puppeteer.launch({
  headless: true,
  args: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage', '--font-render-hinting=none']
});

const page = await browser.newPage();
await page.setViewport({ width: 1440, height: 900, deviceScaleFactor: 1.5 });

// Load fonts
await page.goto(url, { waitUntil: 'networkidle2', timeout: 30000 });
await new Promise(r => setTimeout(r, 2000));

// 1. Hero
await page.evaluate(() => window.scrollTo(0, 0));
await new Promise(r => setTimeout(r, 500));
await page.screenshot({ path: path.join(__dirname, 'shot-hero.jpg'), type: 'jpeg', quality: 90, clip: { x: 0, y: 0, width: 1440, height: 900 } });
console.log('✓ hero');

// 2. Mission section
const missionEl = await page.$('#mission');
const missionBox = await missionEl.boundingBox();
await page.evaluate((y) => window.scrollTo({ top: y - 80, behavior: 'instant' }), missionBox.y);
await new Promise(r => setTimeout(r, 400));
await page.screenshot({ path: path.join(__dirname, 'shot-mission.jpg'), type: 'jpeg', quality: 90, clip: { x: 0, y: missionBox.y - 80, width: 1440, height: 900 } });
console.log('✓ mission');

// 3. Organizations
const orgEl = await page.$('#organizations');
const orgBox = await orgEl.boundingBox();
await page.evaluate((y) => window.scrollTo({ top: y - 80, behavior: 'instant' }), orgBox.y);
await new Promise(r => setTimeout(r, 400));
await page.screenshot({ path: path.join(__dirname, 'shot-orgs.jpg'), type: 'jpeg', quality: 90, clip: { x: 0, y: orgBox.y - 80, width: 1440, height: 900 } });
console.log('✓ organizations');

// 4. Impact + Armenia
const impactEl = await page.$('#impact');
const impactBox = await impactEl.boundingBox();
await page.evaluate((y) => window.scrollTo({ top: y - 80, behavior: 'instant' }), impactBox.y);
await new Promise(r => setTimeout(r, 400));
await page.screenshot({ path: path.join(__dirname, 'shot-impact.jpg'), type: 'jpeg', quality: 90, clip: { x: 0, y: impactBox.y - 80, width: 1440, height: 900 } });
console.log('✓ impact');

// 5. Contact
const contactEl = await page.$('#contact');
const contactBox = await contactEl.boundingBox();
await page.evaluate((y) => window.scrollTo({ top: y - 80, behavior: 'instant' }), contactBox.y);
await new Promise(r => setTimeout(r, 400));
await page.screenshot({ path: path.join(__dirname, 'shot-contact.jpg'), type: 'jpeg', quality: 90, clip: { x: 0, y: contactBox.y - 80, width: 1440, height: 900 } });
console.log('✓ contact');

await browser.close();
console.log('done');

File: screenshot.py
import asyncio
from playwright.async_api import async_playwright
import os

html_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'index.html'))
url = f'file://{html_path}'
out_dir = os.path.dirname(__file__)

async def shot():
    async with async_playwright() as pw:
        browser = await pw.chromium.launch(headless=True, args=['--no-sandbox'])
        page = await browser.new_page(viewport={'width': 1440, 'height': 900})
        await page.goto(url, wait_until='networkidle')
        await asyncio.sleep(2)

        sections = [
            ('hero', '#home'),
            ('mission', '#mission'),
            ('orgs', '#organizations'),
            ('impact', '#impact'),
            ('contact', '#contact'),
        ]

        for name, sel in sections:
            el = await page.query_selector(sel)
            # Scroll the element into view
            await el.scroll_into_view_if_needed()
            await asyncio.sleep(0.8)
            await page.screenshot(
                path=f'{out_dir}/shot-{name}.jpg',
                type='jpeg',
                quality=92,
            )
            print(f'✓ {name}')

        await browser.close()
        print('done')

asyncio.run(shot())
